src/liballoc/string.rs

```diff
@@ -2031,19 +2031,23 @@ impl ops::DerefMut for String {
 
 /// An error when parsing a `String`.
 ///
+/// As of Rust 1.26, this is a type alias for [`!`].  Code that doesn't need to
+/// support compilation with older compiler versions should just use that type
+/// directly; this alias will be deprecated in the future.
+///
 /// This `enum` is slightly awkward: it will never actually exist. This error is
 /// part of the type signature of the implementation of [`FromStr`] on
 /// [`String`]. The return type of [`from_str`], requires that an error be
 /// defined, but, given that a [`String`] can always be made into a new
 /// [`String`] without error, this type will never actually be returned. As
 /// such, it is only here to satisfy said signature, and is useless otherwise.
 ///
+/// [`!`]: ../../std/primitive.never.html
 /// [`FromStr`]: ../../std/str/trait.FromStr.html
 /// [`String`]: struct.String.html
 /// [`from_str`]: ../../std/str/trait.FromStr.html#tymethod.from_str
 #[stable(feature = "str_parse_error", since = "1.5.0")]
-#[derive(Copy)]
-pub enum ParseError {}
+pub type ParseError = !;
```

Comment:
You can't do this because people can `impl` it. You should (almost) *never* change **stable** *nominal* types.

Reply:
While there's no problem with `impl MyTrait for ParseError` (you can `impl` through type aliases and there are no stable impls for `!`), @eddyb pointed out on IRC that `impl MyTrait for fn()->!` and `impl MyTrait for fn()->ParseError` are allowed on stable, so this may not be allowed even though the break found by crater is allowed inference/coercion changes (it's possible to write in a way that works before&after by specifying the type explicitly).

Reply:
I was informed on IRC that this could be fine because it's the *only* such type and you can't `impl` trait for `!` on stable, but that's not enough, e.g.:
```rust
trait Trait {}
impl Trait for fn() -> ! {}
impl Trait for fn() -> ParseError {}
```